tests/test_test_tools/test_test_contract.py

```diff
@@ -0,0 +1,225 @@
+    @mock.patch.object(FetchAIApi, "get_transaction_receipt", return_value="tx_receipt")
+    @mock.patch.object(FetchAIApi, "is_transaction_settled", return_value=True)
+    def test_sign_send_confirm_receipt_multisig_transaction(
+        self,
+        is_transaction_settled_mock,
+        get_transaction_receipt_mock,
+        send_signed_transaction_mock,
+        sign_transaction_mock,
+    ):
+        """Test the sign_send_confirm_receipt_multisig_transaction static method."""
+        sleep_time = 0
+        tx = "tx"
+        result = self.sign_send_confirm_receipt_multisig_transaction(
+            tx, self.ledger_api, [self.deployer_crypto], sleep_time=sleep_time
+        )
+        assert result == "tx_receipt"
+        is_transaction_settled_mock.assert_called_once_with("tx_receipt")
+        get_transaction_receipt_mock.assert_called_with("tx_digest")
+        send_signed_transaction_mock.assert_called_once_with(tx)
+        sign_transaction_mock.assert_called_once_with(tx)
+
+    @mock.patch.object(FetchAICrypto, "sign_transaction", return_value="tx")
+    @mock.patch.object(FetchAIApi, "send_signed_transaction", return_value=None)
+    @mock.patch.object(FetchAIApi, "get_transaction_receipt")
+    @mock.patch.object(FetchAIApi, "is_transaction_settled")
+    def test_sign_send_confirm_receipt_multisig_transaction_digest_not_found(
+        self,
+        is_transaction_settled_mock,
+        get_transaction_receipt_mock,
+        send_signed_transaction_mock,
+        sign_transaction_mock,
+    ):
+        """Test the sign_send_confirm_receipt_multisig_transaction static method: digest not found."""
+        tx = "tx"
+        with pytest.raises(ValueError) as e:
+            self.sign_send_confirm_receipt_multisig_transaction(
+                tx, self.ledger_api, [self.deployer_crypto],
+            )
+            assert str(e) == "Transaction digest not found!"
+        is_transaction_settled_mock.assert_not_called()
+        get_transaction_receipt_mock.assert_not_called()
+        send_signed_transaction_mock.assert_called_once_with(tx)
+        sign_transaction_mock.assert_called_once_with(tx)
+
+    @mock.patch.object(FetchAICrypto, "sign_transaction", return_value="tx")
+    @mock.patch.object(FetchAIApi, "send_signed_transaction", return_value="tx_digest")
+    @mock.patch.object(FetchAIApi, "get_transaction_receipt", return_value=None)
+    @mock.patch.object(FetchAIApi, "is_transaction_settled")
+    def test_sign_send_confirm_receipt_multisig_transaction_receipt_not_found(
+        self,
+        is_transaction_settled_mock,
+        get_transaction_receipt_mock,
+        send_signed_transaction_mock,
+        sign_transaction_mock,
+    ):
+        """Test the sign_send_confirm_receipt_multisig_transaction static method: receipt not found."""
+        tx = "tx"
+        sleep_time = 0
+        with pytest.raises(ValueError) as e:
```

Comment:
```suggestion
        with pytest.raises(ValueError, match="Transaction receipt not found!") as e:
```

Reply:
`Regex pattern 'Transaction digest not found!' does not match 'Transaction receipt not found!'.`
`match` seems to be a regex. So I'll leave assertion of exact exception message that I have below.